resources/check-layout-th.js

```diff
@@ -258,4 +250,17 @@ window.checkLayout = async function(selectorList, callDone = true)
         done();
 };
 
+window.checkLayout = function(selectorList, callDone = true) {
+    // This is a reserved variable not intended to be used by browsers.
+    // Instead, polyfills can set this variable to give themselves time
+    // to apply changes before proceeding with assertions about the layout.
+    const delay = window.CHECK_LAYOUT_DELAY_MS;
+    if (typeof delay === 'number') {
+        console.log(`Waiting ${delay}ms before checking layout...`);
+        step_timeout(() => _checkLayout(selectorList, callDone), delay)
+    } else {
+        _checkLayout(selectorList, callDone)
+    }
+}
```

Comment:
Since this is no longer an `async` function, in cases where `window.CHECK_LAYOUT_DELAY_MS` is set and this `checkLayout` is called twice, there might be a race condition now where changes are made after the first `checkLayout` call, but before the test assertions run.

Is there a reason to prefer this approach to the `async/await` option?

Reply:
It seemed to me like avoiding `async` was less disruptive than modifying multiple call sites, but I see what you mean. I'll go back to the `async` approach

Reply:
I agree with you that this is (much) less disruptive, but I'm worried we might cause unintentional side-effects if we add a delay without accounting for it where the fn is called.

Reply:
New implementation in https://github.com/oddbird/wpt/pull/1/files/7d13b39469b4824d306b94824555cf251f06a4c2#diff-b8002771420c10969dc9ed103bcf6d7efb79165532c02a97b0b19218b21a0bc0